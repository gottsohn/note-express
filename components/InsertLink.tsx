import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
import Modal from 'react-native-modal';

interface Props {
  color: string;
  placeholderColor: string;
  backgroundColor: string;
  onDone: (urlParam: URLParam) => void,
}

interface URLParam {
  title: string;
  url: string;
}

interface State {
  isModalVisible: boolean;
}

export class InsertLinkModal extends Component<Props, State> {
  private title: string = '';
  private url: string = '';
  constructor(props: Props) {
    super(props);
    this.state = {
      isModalVisible: false,
    };
  }

  public setModalVisible = (isModalVisible: boolean) => {
    this.setState({ isModalVisible });
  };

  private setTitle = (title: string) => {
    this.title = title;
  };

  private setURL = (url: string) => {
    this.url = url;
  };

  private onDone = () => {
    const title = this.title;
    const url = this.url;
    this.setModalVisible(false);
    this.props.onDone({ title, url });
  }

  render() {
    const { isModalVisible } = this.state;
    const { color, placeholderColor, backgroundColor } = this.props;
    return (
      <Modal
        isVisible={isModalVisible}
        backdropColor={color}
        backdropOpacity={0.3}
        onBackdropPress={() => this.setModalVisible(false)}>
        <View style={[styles.dialog, { backgroundColor }]}>
          <View style={styles.linkTitle}>
            <Text style={{ color }}>Insert Link</Text>
          </View>
          <View style={styles.item}>
            <TextInput
              style={[styles.input, { color }]}
              placeholderTextColor={placeholderColor}
              placeholder={'title'}
              onChangeText={(text) => this.setTitle(text)}
            />
          </View>
          <View style={styles.item}>
            <TextInput
              style={[styles.input, { color }]}
              placeholderTextColor={placeholderColor}
              placeholder="http(s)://"
              onChangeText={(text) => this.setURL(text)}
            />
          </View>
          <View style={styles.buttonView}>
            <TouchableOpacity style={styles.btn} onPress={() => this.setModalVisible(false)}>
              <Text style={styles.text}>Cancel</Text>
            </TouchableOpacity>
            <TouchableOpacity style={styles.btn} onPress={this.onDone}>
              <Text style={styles.text}>OK</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    );
  }
}

const styles = StyleSheet.create({
  item: {
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderColor: '#e8e8e8',
    flexDirection: 'row',
    height: 40,
    alignItems: 'center',
    paddingHorizontal: 15,
  },
  input: {
    flex: 1,
    height: 40,
  },
  linkTitle: {
    height: 36,
    justifyContent: 'center',
    alignItems: 'center',
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderColor: '#b3b3b3',
  },
  dialog: {
    borderRadius: 8,
    marginHorizontal: 40,
    paddingHorizontal: 10,
  },

  buttonView: {
    flexDirection: 'row',
    height: 36,
    paddingVertical: 4,
  },
  btn: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: '#286ab2',
  },
});