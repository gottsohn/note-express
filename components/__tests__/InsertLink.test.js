import * as React from 'react';
import renderer from 'react-test-renderer';

import { InsertLinkModal } from '../InsertLink';

const props = {
  color: '#111',
  backgroundColor: '#fff',
  placeholderColor: '#777',
  onDone: (url) => url,
};

describe('InsertLinkModal', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<InsertLinkModal {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
})
