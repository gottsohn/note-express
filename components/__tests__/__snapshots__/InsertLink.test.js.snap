// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsertLinkModal renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#111",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 8,
            "marginHorizontal": 40,
            "paddingHorizontal": 10,
          },
          Object {
            "backgroundColor": "#fff",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0.5,
            "borderColor": "#b3b3b3",
            "height": 36,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#111",
            }
          }
        >
          Insert Link
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0.5,
            "borderColor": "#e8e8e8",
            "flexDirection": "row",
            "height": 40,
            "paddingHorizontal": 15,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="title"
          placeholderTextColor="#777"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "flex": 1,
                "height": 40,
              },
              Object {
                "color": "#111",
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0.5,
            "borderColor": "#e8e8e8",
            "flexDirection": "row",
            "height": 40,
            "paddingHorizontal": 15,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="http(s)://"
          placeholderTextColor="#777"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "flex": 1,
                "height": 40,
              },
              Object {
                "color": "#111",
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 36,
            "paddingVertical": 4,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#286ab2",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#286ab2",
              }
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
